### PR TITLE
Add test for footer and categoryController

### PR DESCRIPTION
**Pull request by zenithyap**
Create test for footer component and categoryController.

### CODE REVIEW

File: controllers/categoryController.test.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        expect(res.status).toHaveBeenCalledWith(401);
+        expect(res.send).toHaveBeenCalledWith({ message: "Name is required" });
+    })
+
+    test('should return 200 if catergory exists', async () => {
+        req.body = { name: "test" };
+        categoryModel.findOne.mockResolvedValue({ name: "test" });
+
+        await createCategoryController(req, res);
+        expect(res.status).toHaveBeenCalledWith(200);
```
Comment (eugenetangkj) [inline controllers/categoryController.test.js]: Status code `200` means success or OK. It should not have returned 200 if category exists. Furthermore, why is `success` true when category exists? Shouldn't it be false?

Refer [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) for more information.

File: controllers/categoryController.test.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    updateCategoryController, 
+    categoryControlller, 
+    singleCategoryController, 
+    deleteCategoryCOntroller
+} from "./categoryController";
+
+jest.mock("../models/categoryModel");
+jest.mock("slugify");
+
+describe('category controller', () => {
```
Comment (eugenetangkj) [inline controllers/categoryController.test.js:12]: Might be better to split into different `describe` blocks. For example, one describe block for `createCategoryController` and another for `updateCategoryController` etc.

File: controllers/categoryController.test.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            success: true,
+            message: 'new category created',
+            category: { name: 'test', slug: 'test-slug' }
+        });
+    });
+
+    test('should return 500 and error while creating category', async () => {
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+        req.body = { name: 'test' };
+        categoryModel.findOne = jest.fn().mockRejectedValue(new Error('Database error'));
```
Comment (eugenetangkj) [inline controllers/categoryController.test.js]: Would it be better to make the error a variable, so you can check that that exact error is being called later?

In other words:
```
const error = new Error('Database error');
categoryModel.findOne = jest.fn().mockRejectedValue(error);
```
Then, verify whether `error` is called in `console.log` etc.

File: controllers/categoryController.test.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        expect(res.status).toHaveBeenCalledWith(201);
+        expect(res.send).toHaveBeenCalledWith({
+            success: true,
+            message: 'new category created',
+            category: { name: 'test', slug: 'test-slug' }
+        });
+    });
+
+    test('should return 500 and error while creating category', async () => {
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
```
Comment (eugenetangkj) [inline controllers/categoryController.test.js:99]: This `consoleLogSpy` does not seem to be used in any assertions. Did you forget to check it?

File: controllers/categoryController.test.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith({
+            success: true,
+            message: "All Categories List",
+            category: {},
+        })
+    });
+
+    test('should return 500 while getting categories', async () => {
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
```
Comment (eugenetangkj) [inline controllers/categoryController.test.js]: Same issue as above about `consoleLogSpy`

**Review by eugenetangkj — CHANGES_REQUESTED** [review]
Great effort! Left some comments for your perusal. 😄

**Comment by zenithyap** [comment]
Thanks Eugene, I have updated the code. Let me know if there are further changes to be made. Thanks!

**Review by eugenetangkj — APPROVED** [review]

**Comment by eugenetangkj** [comment]
> Thanks Eugene, I have updated the code. Let me know if there are further changes to be made. Thanks!

Thanks! Looks good to merge. Great effort 😄

**Pull request merged**